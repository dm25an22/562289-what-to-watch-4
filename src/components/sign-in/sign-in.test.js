import React from "react";
import renderer from "react-test-renderer";
import SignIn from "./sign-in.jsx";
import {Router} from "react-router-dom";
import {history} from "../../history";

it(`Render SignIn default`, () => {
  const tree = renderer.create(
      <Router history={history}>
        <SignIn
          handleSubmit={() => {}}
          loginClass={`loginClass`}
          passwordClass={`passwordClass`}
          resetLoginClassName={() => {}}
          resetPasswordClassName={() => {}}
          invalidLogin={false}
          badRequest={false}
          invalidPassword={false}
          getLoginValue={() => {}}
          getPasswordValue={() => {}}
        />
      </Router>
  ).toJSON();

  expect(tree).toMatchSnapshot();
});

it(`Render SignIn with invalid login`, () => {
  const tree = renderer.create(
      <Router history={history}>
        <SignIn
          handleSubmit={() => {}}
          loginClass={`loginClass-error`}
          passwordClass={`passwordClass`}
          resetLoginClassName={() => {}}
          resetPasswordClassName={() => {}}
          invalidLogin={true}
          badRequest={false}
          invalidPassword={false}
          getLoginValue={() => {}}
          getPasswordValue={() => {}}
        />
      </Router>
  ).toJSON();

  expect(tree).toMatchSnapshot();
});

it(`Render SignIn with invalid password`, () => {
  const tree = renderer.create(
      <Router history={history}>
        <SignIn
          handleSubmit={() => {}}
          loginClass={`loginClass-error`}
          passwordClass={`passwordClass-error`}
          resetLoginClassName={() => {}}
          resetPasswordClassName={() => {}}
          invalidLogin={false}
          badRequest={false}
          invalidPassword={true}
          getLoginValue={() => {}}
          getPasswordValue={() => {}}
        />
      </Router>
  ).toJSON();

  expect(tree).toMatchSnapshot();
});


it(`Render SignIn with bad request`, () => {
  const tree = renderer.create(
      <Router history={history}>
        <SignIn
          handleSubmit={() => {}}
          loginClass={`loginClass-error`}
          passwordClass={`passwordClass`}
          resetLoginClassName={() => {}}
          resetPasswordClassName={() => {}}
          invalidLogin={false}
          badRequest={true}
          invalidPassword={false}
          getLoginValue={() => {}}
          getPasswordValue={() => {}}
        />
      </Router>
  ).toJSON();

  expect(tree).toMatchSnapshot();
});
