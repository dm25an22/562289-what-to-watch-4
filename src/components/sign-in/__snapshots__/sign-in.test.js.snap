// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Render SignIn default 1`] = `
<div
  className="user-page"
>
  <header
    className="page-header user-page__head"
  >
    <div
      className="logo"
    >
      <a
        className="logo__link undefined"
        href="/"
        onClick={[Function]}
      >
        <span
          className="logo__letter logo__letter--1"
        >
          W
        </span>
        <span
          className="logo__letter logo__letter--2"
        >
          T
        </span>
        <span
          className="logo__letter logo__letter--3"
        >
          W
        </span>
      </a>
    </div>
    <h1
      className="page-title user-page__title"
    >
      Sign in
    </h1>
  </header>
  <div
    className="sign-in user-page__content"
  >
    <form
      action="#"
      className="sign-in__form"
      noValidate={true}
      onSubmit={[Function]}
    >
      <div
        className="sign-in__message"
      />
      <div
        className="sign-in__fields"
      >
        <div
          className="loginClass"
        >
          <input
            className="sign-in__input"
            id="user-email"
            name="user-email"
            onChange={[Function]}
            onFocus={[Function]}
            placeholder="Email address"
            type="email"
          />
          <label
            className="sign-in__label visually-hidden"
            htmlFor="user-email"
          >
            Email address
          </label>
        </div>
        <div
          className="passwordClass"
        >
          <input
            className="sign-in__input"
            id="user-password"
            name="user-password"
            onChange={[Function]}
            onFocus={[Function]}
            placeholder="Password"
            type="password"
          />
          <label
            className="sign-in__label visually-hidden"
            htmlFor="user-password"
          >
            Password
          </label>
        </div>
      </div>
      <div
        className="sign-in__submit"
      >
        <button
          className="sign-in__btn"
          type="submit"
        >
          Sign in
        </button>
      </div>
    </form>
  </div>
  <footer
    className="page-footer"
  >
    <div
      className="logo"
    >
      <a
        className="logo__link logo__link--light"
        href="/"
        onClick={[Function]}
      >
        <span
          className="logo__letter logo__letter--1"
        >
          W
        </span>
        <span
          className="logo__letter logo__letter--2"
        >
          T
        </span>
        <span
          className="logo__letter logo__letter--3"
        >
          W
        </span>
      </a>
    </div>
    <div
      className="copyright"
    >
      <p>
        © 2019 What to watch Ltd.
      </p>
    </div>
  </footer>
</div>
`;

exports[`Render SignIn with bad request 1`] = `
<div
  className="user-page"
>
  <header
    className="page-header user-page__head"
  >
    <div
      className="logo"
    >
      <a
        className="logo__link undefined"
        href="/"
        onClick={[Function]}
      >
        <span
          className="logo__letter logo__letter--1"
        >
          W
        </span>
        <span
          className="logo__letter logo__letter--2"
        >
          T
        </span>
        <span
          className="logo__letter logo__letter--3"
        >
          W
        </span>
      </a>
    </div>
    <h1
      className="page-title user-page__title"
    >
      Sign in
    </h1>
  </header>
  <div
    className="sign-in user-page__content"
  >
    <form
      action="#"
      className="sign-in__form"
      noValidate={true}
      onSubmit={[Function]}
    >
      <div
        className="sign-in__message"
      >
        <p>
          We can’t recognize this email 
          <br />
           and password combination. Please try again.
        </p>
      </div>
      <div
        className="sign-in__fields"
      >
        <div
          className="loginClass-error"
        >
          <input
            className="sign-in__input"
            id="user-email"
            name="user-email"
            onChange={[Function]}
            onFocus={[Function]}
            placeholder="Email address"
            type="email"
          />
          <label
            className="sign-in__label visually-hidden"
            htmlFor="user-email"
          >
            Email address
          </label>
        </div>
        <div
          className="passwordClass"
        >
          <input
            className="sign-in__input"
            id="user-password"
            name="user-password"
            onChange={[Function]}
            onFocus={[Function]}
            placeholder="Password"
            type="password"
          />
          <label
            className="sign-in__label visually-hidden"
            htmlFor="user-password"
          >
            Password
          </label>
        </div>
      </div>
      <div
        className="sign-in__submit"
      >
        <button
          className="sign-in__btn"
          type="submit"
        >
          Sign in
        </button>
      </div>
    </form>
  </div>
  <footer
    className="page-footer"
  >
    <div
      className="logo"
    >
      <a
        className="logo__link logo__link--light"
        href="/"
        onClick={[Function]}
      >
        <span
          className="logo__letter logo__letter--1"
        >
          W
        </span>
        <span
          className="logo__letter logo__letter--2"
        >
          T
        </span>
        <span
          className="logo__letter logo__letter--3"
        >
          W
        </span>
      </a>
    </div>
    <div
      className="copyright"
    >
      <p>
        © 2019 What to watch Ltd.
      </p>
    </div>
  </footer>
</div>
`;

exports[`Render SignIn with invalid login 1`] = `
<div
  className="user-page"
>
  <header
    className="page-header user-page__head"
  >
    <div
      className="logo"
    >
      <a
        className="logo__link undefined"
        href="/"
        onClick={[Function]}
      >
        <span
          className="logo__letter logo__letter--1"
        >
          W
        </span>
        <span
          className="logo__letter logo__letter--2"
        >
          T
        </span>
        <span
          className="logo__letter logo__letter--3"
        >
          W
        </span>
      </a>
    </div>
    <h1
      className="page-title user-page__title"
    >
      Sign in
    </h1>
  </header>
  <div
    className="sign-in user-page__content"
  >
    <form
      action="#"
      className="sign-in__form"
      noValidate={true}
      onSubmit={[Function]}
    >
      <div
        className="sign-in__message"
      >
        <p>
          Please enter a valid email address
        </p>
      </div>
      <div
        className="sign-in__fields"
      >
        <div
          className="loginClass-error"
        >
          <input
            className="sign-in__input"
            id="user-email"
            name="user-email"
            onChange={[Function]}
            onFocus={[Function]}
            placeholder="Email address"
            type="email"
          />
          <label
            className="sign-in__label visually-hidden"
            htmlFor="user-email"
          >
            Email address
          </label>
        </div>
        <div
          className="passwordClass"
        >
          <input
            className="sign-in__input"
            id="user-password"
            name="user-password"
            onChange={[Function]}
            onFocus={[Function]}
            placeholder="Password"
            type="password"
          />
          <label
            className="sign-in__label visually-hidden"
            htmlFor="user-password"
          >
            Password
          </label>
        </div>
      </div>
      <div
        className="sign-in__submit"
      >
        <button
          className="sign-in__btn"
          type="submit"
        >
          Sign in
        </button>
      </div>
    </form>
  </div>
  <footer
    className="page-footer"
  >
    <div
      className="logo"
    >
      <a
        className="logo__link logo__link--light"
        href="/"
        onClick={[Function]}
      >
        <span
          className="logo__letter logo__letter--1"
        >
          W
        </span>
        <span
          className="logo__letter logo__letter--2"
        >
          T
        </span>
        <span
          className="logo__letter logo__letter--3"
        >
          W
        </span>
      </a>
    </div>
    <div
      className="copyright"
    >
      <p>
        © 2019 What to watch Ltd.
      </p>
    </div>
  </footer>
</div>
`;

exports[`Render SignIn with invalid password 1`] = `
<div
  className="user-page"
>
  <header
    className="page-header user-page__head"
  >
    <div
      className="logo"
    >
      <a
        className="logo__link undefined"
        href="/"
        onClick={[Function]}
      >
        <span
          className="logo__letter logo__letter--1"
        >
          W
        </span>
        <span
          className="logo__letter logo__letter--2"
        >
          T
        </span>
        <span
          className="logo__letter logo__letter--3"
        >
          W
        </span>
      </a>
    </div>
    <h1
      className="page-title user-page__title"
    >
      Sign in
    </h1>
  </header>
  <div
    className="sign-in user-page__content"
  >
    <form
      action="#"
      className="sign-in__form"
      noValidate={true}
      onSubmit={[Function]}
    >
      <div
        className="sign-in__message"
      >
        <p>
          Your password must contain at least one character
        </p>
      </div>
      <div
        className="sign-in__fields"
      >
        <div
          className="loginClass-error"
        >
          <input
            className="sign-in__input"
            id="user-email"
            name="user-email"
            onChange={[Function]}
            onFocus={[Function]}
            placeholder="Email address"
            type="email"
          />
          <label
            className="sign-in__label visually-hidden"
            htmlFor="user-email"
          >
            Email address
          </label>
        </div>
        <div
          className="passwordClass-error"
        >
          <input
            className="sign-in__input"
            id="user-password"
            name="user-password"
            onChange={[Function]}
            onFocus={[Function]}
            placeholder="Password"
            type="password"
          />
          <label
            className="sign-in__label visually-hidden"
            htmlFor="user-password"
          >
            Password
          </label>
        </div>
      </div>
      <div
        className="sign-in__submit"
      >
        <button
          className="sign-in__btn"
          type="submit"
        >
          Sign in
        </button>
      </div>
    </form>
  </div>
  <footer
    className="page-footer"
  >
    <div
      className="logo"
    >
      <a
        className="logo__link logo__link--light"
        href="/"
        onClick={[Function]}
      >
        <span
          className="logo__letter logo__letter--1"
        >
          W
        </span>
        <span
          className="logo__letter logo__letter--2"
        >
          T
        </span>
        <span
          className="logo__letter logo__letter--3"
        >
          W
        </span>
      </a>
    </div>
    <div
      className="copyright"
    >
      <p>
        © 2019 What to watch Ltd.
      </p>
    </div>
  </footer>
</div>
`;
